Fixed description of scheduling semantics. (#2400)
ros2/ros2_documentation: source/Concepts/About-Executors.rst

By invoking ``spin()`` of the Executor instance, the current thread starts querying the rcl and middleware layers for incoming messages and other events and calls the corresponding callback functions until the node shuts down.
In order not to counteract the QoS settings of the middleware, an incoming message is not stored in a queue on the Client Library layer but kept in the middleware until it is taken for processing by a callback function.
(This is a crucial difference to ROS 1.)

> A *wait set* is used to inform the Executor about available messages on the middleware layer and on timer events on the rcl layer, with one binary flag per queue.

Reviewer: ```suggestion
A *wait set* is used to inform the Executor about available messages on the middleware layer, with one binary flag per queue.
The *wait set* is also used to detect when timers expire.
```

The "with one binary flag per queue" statement doesn't really apply to timers, so I suggested we separate the statements.

The wait set mechanism reports only very little information about these queues to the Executor.
In detail, it only reports whether there are any messages for a certain topic or not.
The Executor uses this information to process the messages (including services and actions) in a round-robin fashion - but not in FIFO order.
The following flow diagram visualizes this scheduling semantics.

.. image:: images/executors_scheduling_semantics.png

This semantics was first described in a `paper by Casini et al. at ECRTS 2019 <https://drops.dagstuhl.de/opus/volltexte/2019/10743/pdf/LIPIcs-ECRTS-2019-6.pdf>`_.

> (Note: The paper also explains that timer events are prioritized over all other messages. This prioritization was removed in Eloquent.)

Reviewer: Can you cite that change in eloquent? Maybe a pr?
